src/assets/wise5/authoringTool/node/node-authoring/node-authoring.component.ts

```diff
@@ -292,37 +292,56 @@ export class NodeAuthoringComponent implements OnInit {
     this.hideComponentAuthoring();
   }
 
-  protected deleteButtonClicked(): void {
+  protected deleteComponents(): void {
     this.scrollToTopOfPage();
     this.hideComponentAuthoring();
-
-    let confirmMessage = '';
     const selectedComponentNumbersAndTypes = this.getSelectedComponentNumbersAndTypes();
-    if (selectedComponentNumbersAndTypes.length == 1) {
-      confirmMessage = $localize`Are you sure you want to delete this component?\n`;
-    } else if (selectedComponentNumbersAndTypes.length > 1) {
-      confirmMessage = $localize`Are you sure you want to delete these components?\n`;
-    }
-    for (let c = 0; c < selectedComponentNumbersAndTypes.length; c++) {
-      confirmMessage += '\n' + selectedComponentNumbersAndTypes[c];
-    }
-    if (confirm(confirmMessage)) {
-      const selectedComponents = this.getSelectedComponentIds();
-      const data = {
-        componentsDeleted: this.getComponentObjectsForEventData(selectedComponents)
-      };
-      for (const componentId of selectedComponents) {
+    if (this.confirmDeleteComponent(selectedComponentNumbersAndTypes)) {
+      const selectedComponentIds = this.getSelectedComponentIds();
+      const componentIdAndTypes = this.getComponentObjectsForEventData(selectedComponentIds);
+      for (const componentId of selectedComponentIds) {
         this.projectService.deleteComponent(this.nodeId, componentId);
       }
-      this.saveEvent('componentDeleted', 'Authoring', data);
-      this.checkIfNeedToShowNodeSaveOrNodeSubmitButtons();
-      this.projectService.saveProject();
+      this.afterDeleteComponent(componentIdAndTypes);
     }
     this.turnOffInsertComponentMode();
-    this.clearComponentsToChecked();
     this.showComponentAuthoring();
   }
 
+  protected deleteComponent(event: any, componentNumber: number, component: any): void {
+    event.stopPropagation();
+    if (this.confirmDeleteComponent([`${componentNumber}. ${component.type}`])) {
+      const componentIdAndTypes = this.getComponentObjectsForEventData([component.id]);
+      this.projectService.deleteComponent(this.nodeId, component.id);
+      this.afterDeleteComponent(componentIdAndTypes);
+    }
+  }
+
+  private confirmDeleteComponent(componentLabels: string[]): boolean {
+    let confirmMessage = '';
+    if (componentLabels.length === 1) {
+      confirmMessage = $localize`Are you sure you want to delete this component?\n`;
+    } else if (componentLabels.length > 1) {
+      confirmMessage = $localize`Are you sure you want to delete these components?\n`;
+    }
```

Comment:
Use ternary statement for brevity/clarity? We'll never call this function with length = 0.
```suggestion
    let confirmMessage =
      componentLabels.length === 1
        ? 'Are you sure you want to delete this component?\n'
        : 'Are you sure you want to delete these components?\n';
```